backend/controller/scaling/k8sscaling/k8s_scaling.go

```diff
@@ -0,0 +1,85 @@
+package k8sscaling
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/TBD54566975/ftl/backend/controller/leases"
+	"github.com/TBD54566975/ftl/backend/controller/scaling"
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+var _ = (scaling.RunnerScaling)(NewK8sScaling)
+
+func NewK8sScaling(ctx context.Context, controller url.URL, leaser leases.Leaser) error {
+
+	logger := log.FromContext(ctx).Scope("K8sScaling")
+	ctx = log.ContextWithLogger(ctx, logger)
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+
+	if err != nil {
+		// if we're not in a cluster, use the kubeconfig
+		config, err = clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+
+	if err != nil {
+		panic(err.Error())
+	}
+
+	namespace, err := getCurrentNamespace()
+	if err != nil {
+		// Nothing we can do here, if we don't have a namespace we have no runners
+		return fmt.Errorf("failed to get current namespace: %w", err)
+	}
+	logger.Infof("using namespace %s", namespace)
+	deploymentReconciler := &DeploymentProvisioner{
+		Client:       clientset,
+		Namespace:    namespace,
+		KnownModules: map[string]string{},
+		FTLEndpoint:  controller.String(),
+	}
+	scaling.BeginGrpcScaling(ctx, controller, leaser, deploymentReconciler.HandleSchemaChange)
+	return nil
+}
+
+func getCurrentNamespace() (string, error) {
+	namespaceFile := "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+	if _, err := os.Stat(namespaceFile); err == nil {
```

Comment:
Typically there's no need to stat then read, just read it and check for ErrNotFound.